### PR TITLE
HDDS-6118. Documentation missing settings to secure S3g with Kerberos

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

1. Add missing S3G settings.

2. Optimized for ZH document content translation.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/projects/HDDS/issues/HDDS-6118

## How was this patch tested?

Local hugo test.

### CODE REVIEW

File: hadoop-hdds/docs/content/security/SecureOzone.md [hunk 1]
```diff
@@ -166,6 +166,14 @@ All these settings should be made in ozone-site.xml.
           </tr>
         </thead>
         <tbody>
+          <tr>
+            <td>ozone.s3g.kerberos.principal</th>
+            <td>S3 Gateway principal. <br/> e.g. s3g/_host@REAML</td>
```
Comment (adoroszlai) [inline hadoop-hdds/docs/content/security/SecureOzone.md:171]: Typo:

```suggestion
            <td>S3 Gateway principal. <br/> e.g. s3g/_HOST@REALM</td>
```

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 2]
```diff
@@ -146,12 +146,20 @@ ozone-site.xml 中应进行如下配置：
         </thead>
         <tbody>
           <tr>
-            <td>ozone.s3g.authentication.kerberos.principal</th>
-            <td>S3 网关服务主体，例如：HTTP/_HOST@EXAMPLE.COM</td>
+            <td>ozone.s3g.kerberos.principal</th>
+            <td>S3 网关主体，例如：s3g/_host@REAML</td>
```
Comment (adoroszlai) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:150]: Same typo here:

```suggestion
            <td>S3 网关主体，例如：s3g/_HOST@REALM</td>
```

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 3]
```diff
@@ -146,12 +146,20 @@ ozone-site.xml 中应进行如下配置：
...
           </tr>
           <tr>
-            <td>ozone.s3g.keytab.file</th>
-            <td>S3 网关使用的 keytab 文件</td>
+            <td>ozone.s3g.kerberos.keytab.file</th>
+            <td>S3 网关使用的 keytab 文件，例如：/etc/security/keytabs/s3g.keytab/td>
```
Comment (adoroszlai) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:154]:
```suggestion
            <td>S3 网关使用的 keytab 文件，例如：/etc/security/keytabs/s3g.keytab</td>
```

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @sky76093016 for updating the doc.  I noticed a few typos.

**Comment by sky76093016** [comment]
Thanks to @adoroszlai for reviewing, I will revise it immediately.

File: hadoop-hdds/docs/content/security/SecureOzone.md [hunk 4]
```diff
@@ -168,7 +168,7 @@ All these settings should be made in ozone-site.xml.
         <tbody>
           <tr>
             <td>ozone.s3g.kerberos.principal</th>
-            <td>S3 Gateway principal. <br/> e.g. s3g/_host@REAML</td>
+            <td>S3 Gateway principal. <br/> e.g. s3g/_HOST@REAML</td>
```
Comment (adoroszlai) [inline hadoop-hdds/docs/content/security/SecureOzone.md:171]: Sorry, but `REAML` is still misspelled.

Comment (sky76093016) [reply hadoop-hdds/docs/content/security/SecureOzone.md:171]: Solve it.

**Review by adoroszlai — APPROVED** [review]
Thanks @sky76093016 for updating the patch.  Let's ask someone to review the Chinese part, too.

**Comment by sky76093016** [comment]
@cku328 Please take a look at the Chinese part.

**Comment by cku328** [comment]
Thanks @sky76093016 for fixing on this.
In the same file, can you also help to fix the following typo?
image

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 5]
```diff
@@ -84,12 +84,12 @@ ozone-site.xml 中应进行如下配置：
             <td>SCM 进程使用的 keytab 文件</td>
           </tr>
           <tr>
-            <td>hdds.scm.http.kerberos.principal</th>
-            <td>SCM http 服务主体</td>
+            <td>hdds.scm.http.auth.kerberos.principal</th>
+            <td>SCM http 服务主体，如果为 SCM http 服务器启用了 SPNEGO</td>
```
Comment (cku328) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:88]:
```suggestion
            <td>SCM http 服务主体（当 SCM http 服务器启用了 SPNEGO）</td>
```

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 6]
```diff
@@ -84,12 +84,12 @@ ozone-site.xml 中应进行如下配置：
...
           </tr>
           <tr>
-            <td>hdds.scm.http.kerberos.keytab</th>
-            <td>SCM http 服务使用的 keytab 文件</td>
+            <td>hdds.scm.http.auth.kerberos.keytab</th>
+            <td>SCM http 服务使用的 keytab 文件，如果为 SCM http 服务器启用了 SPNEGO</td>
```
Comment (cku328) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:92]:
```suggestion
            <td>SCM http 服务使用的 keytab 文件（当 SCM http 服务器启用了 SPNEGO）</td>
```

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 7]
```diff
@@ -112,19 +112,19 @@ ozone-site.xml 中应进行如下配置：
...
           </tr>
           <tr>
             <td>ozone.om.kerberos.keytab.file</th>
             <td>OM 进程使用的 keytab 文件</td>
           </tr>
           <tr>
-            <td>ozone.om.http.kerberos.principal</th>
-            <td>OM http 服务主体</td>
+            <td>ozone.om.http.auth.kerberos.principal</th>
+            <td>OM http 服务主体，如果为 om http 服务器启用了 SPNEGO</td>
```
Comment (cku328) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:123]:
```suggestion
            <td>OM http 服务主体（当 om http 服务器启用了 SPNEGO）</td>
```

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 8]
```diff
@@ -112,19 +112,19 @@ ozone-site.xml 中应进行如下配置：
...
           </tr>
           <tr>
-            <td>ozone.om.http.kerberos.keytab</th>
-            <td>OM http 服务使用的 keytab 文件</td>
+            <td>ozone.om.http.auth.kerberos.keytab</th>
+            <td>OM http 服务使用的 keytab 文件，如果为 om http 服务器启用了 SPNEGO</td>
```
Comment (cku328) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:127]:
```suggestion
            <td>OM http 服务使用的 keytab 文件（当 om http 服务器启用了 SPNEGO）</td>
```

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 9]
```diff
@@ -146,12 +146,20 @@ ozone-site.xml 中应进行如下配置：
...
           </tr>
           <tr>
-            <td>ozone.s3g.keytab.file</th>
-            <td>S3 网关使用的 keytab 文件</td>
+            <td>ozone.s3g.kerberos.keytab.file</th>
+            <td>S3 网关使用的 keytab 文件，例如：/etc/security/keytabs/s3g.keytab</td>
+          </tr>
+          <tr>
+            <td>ozone.s3g.http.auth.kerberos.principal</th>
+            <td>S3 网关主体，如果为 S3 网关 http 服务器启用了 SPNEGO，例如：HTTP/_HOST@EXAMPLE.COM</td>
```
Comment (cku328) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:158]:
```suggestion
            <td>S3 网关主体（当 S3 网关 http 服务器启用了 SPNEGO），例如：HTTP/_HOST@EXAMPLE.COM</td>
```

File: hadoop-hdds/docs/content/security/SecureOzone.zh.md [hunk 10]
```diff
@@ -146,12 +146,20 @@ ozone-site.xml 中应进行如下配置：
...
+          </tr>
+          <tr>
+            <td>ozone.s3g.http.auth.kerberos.keytab</th>
+            <td>S3 网关使用的 keytab 文件，如果为 S3 网关 http 服务器启用了 SPNEGO</td>
```
Comment (cku328) [inline hadoop-hdds/docs/content/security/SecureOzone.zh.md:162]:
```suggestion
            <td>S3 网关使用的 keytab 文件（当 S3 网关 http 服务器启用了 SPNEGO）</td>
```

**Comment by sky76093016** [comment]
Thanks for the typo found by @cku328 , let me correct it.

**Review by cku328 — APPROVED** [review]
+1 for the chinese part.
thanks @sky76093016 for updating the patch.

**Comment by adoroszlai** [comment]
Thanks @sky76093016 for the patch, @cku328 for the review.

**Pull request merged**